app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_user_posts_counter
+    user.update(post_counter: user.post.count)
+  end
```

Comment:
- [ ] This should be `author.update(...)`, not `user.update(...)` as a post belongs to an author, not a user.

- **[OPTIONAL]** I'd suggest you add an `after_create` active record callback for this method.